lib/storage/data/file/DataFileStore.js

```diff
@@ -199,6 +199,16 @@ class DataFileStore {
                 return cbOnce(errors.InternalError.customizeDescription(
                     `read stream error: ${err.code}`));
             });
+            dataStream.on('close', () => {
```

Comment:
Does the `close` event never ever trigger in a legitimate end of stream?

Reply:
no, only in case of a close() during a stream processing.